tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -603,9 +770,10 @@ def test_dual_delete_action_missing_aarta(self):
         """
         Test that we raise an exception if the aarta entry for a migrated trigger action is missing
         """
-        migrate_alert_rule(self.metric_alert, self.rpc_user)
-        migrate_metric_data_conditions(self.alert_rule_trigger_critical)
-        migrate_metric_action(self.alert_rule_trigger_action_integration)
```

Comment:
when i was recommending mocks etc, i was thinking you'd create a `setUp` method in these tests that sets up all the data models for the migrations. then you could use a [pytest mock](https://docs.python.org/3/library/unittest.mock.html) to validate things are being called under the hood.

---

The goal for the test refactor should be that we only invoke `migrate_metric_action` and then we can validate that a subset of models were migrated as expected.